tuf/sig.py

```diff
@@ -307,7 +306,12 @@ def verify(signable, role, repository_name='default', threshold=None,
   if threshold is None or threshold <= 0: #pragma: no cover
     raise securesystemslib.exceptions.Error("Invalid threshold: " + repr(threshold))
 
-  return len(set(good_sigs)) >= threshold
+  unique_keys = set()
+  for keyid in good_sigs:
+    key = tuf.keydb.get_key(keyid, repository_name)
+    unique_keys.add(key['keyval']['public'])
```

Comment:
This is great! I believe it is still not entirely robust, as different encodings of the same key, or even the same key with different schemes (e.g., RSA key with different signing/hashing algos) can throw it off.

Reply:
Yes, I think a next step would be to use a consistent representation (modulus and exponent, etc), but I think that would require adding that as a field to securesystemslib. I didn't see anything in the existing key schema could be used.

Reply:
After looking at this some more, I'm not sure if we need to add a consistent representation. The signature will only be verified by a key that has been delegated to, so issues with different encodings, etc will only exist if the delegating party lists the same key in multiple formats. We assume the delegating party is trusted (and if not they could do a much worse attack like reduce the threshold and add their own key). Is there anything I'm missing here?

Reply:
Yeah, we don't have to be this paranoid, but I think better paranoid than sorry. Suppose, for example, the original delegator got fooled, etc. We can file a separate issue for this.

Reply:
In that case, there is also a possibility that the delegator is tricked into changing the `consistent_representation` (or whatever the name would be) field when delegating to a key